src/Orleans.Core/Configuration/OptionLogger/DefaultOptionsFormatter.cs

```diff
@@ -30,21 +32,60 @@ public IEnumerable<string> Format()
             return typeof(T)
                 .GetProperties()
                 .Where(prop => prop.GetGetMethod() != null && prop.GetSetMethod() != null)
-                .Select(FormatProperty)
+                .SelectMany(FormatProperty)
                 .ToList();
         }
 
-        private string FormatProperty(PropertyInfo property)
+        private IEnumerable<string> FormatProperty(PropertyInfo property)
         {
+            var result = new List<string>();
             var name = property.Name;
             var value = property.GetValue(this.options);
             var redactAttribute = property.GetCustomAttribute<RedactAttribute>(inherit: true);
 
-            return OptionFormattingUtilities.Format(
-                name, 
-                redactAttribute != null 
-                    ? redactAttribute.Redact(value)
-                    : value);
+            // If it is a dictionary -> one line per item
+            if (typeof(IDictionary).IsInstanceOfType(value))
+            {
+                var dict = (IDictionary)value;
+                foreach (DictionaryEntry kvp in dict)
+                {
+                    result.Add(OptionFormattingUtilities.Format($"{nameof(dict)}.{kvp.Key}", RedactIfNeeded(redactAttribute, kvp.Value)));
+                }
+            }
+            // if it is a simple collection, flatten everything in one line
+            else if (typeof(ICollection).IsInstanceOfType(value))
+            {
+                var coll = (ICollection)value;
+                if (coll.Count > 0)
+                {
+                    StringBuilder sb = null;
+                    foreach (var item in coll)
+                    {
+                        if (sb == null)
+                            sb = new StringBuilder(RedactIfNeeded(redactAttribute, item).ToString());
```

Comment:
if move the string builder construction out of foreach loop can save one "if" check every looping.

Reply:
I use this trick to now when to add or not a comma. I am playing with ICollection here (and not the generic one) so I don't have a lot of options